pkg/syncer/syncer_process_test.go

```diff
@@ -0,0 +1,859 @@
+			expectActionsOnFrom: []clienttesting.Action{},
+			expectActionsOnTo: []clienttesting.Action{
+				deleteDeploymentAction(
+					"theDeployment",
+					"kcp0124d7647eb6a00b1fcb6f2252201601634989dd79deb7375c373973",
+				),
+			},
+		},
+		"StatusSyncer upsert": {
+			direction:              SyncUp,
+			upstreamLogicalCluster: "root:org:ws",
+			fromNamespace: namespace("kcp0124d7647eb6a00b1fcb6f2252201601634989dd79deb7375c373973", "",
+				map[string]string{
+					"state.internal.workloads.kcp.dev/us-west1": "Sync",
+				},
+				map[string]string{
+					"kcp.dev/namespace-locator": "{\"logical-cluster\":\"root:org:ws\",\"namespace\":\"test\"}",
+				}),
+			gvr: schema.GroupVersionResource{Group: "apps", Version: "v1", Resource: "deployments"},
+			fromResource: changeDeployment(
+				deployment("theDeployment", "kcp0124d7647eb6a00b1fcb6f2252201601634989dd79deb7375c373973", "", map[string]string{
+					"state.internal.workloads.kcp.dev/us-west1": "Sync",
+				}, nil, nil),
+				addDeploymentStatus(appsv1.DeploymentStatus{
+					Replicas: 15,
+				})),
+			toResources: []runtime.Object{
+				deployment("theDeployment", "test", "root:org:ws", map[string]string{
+					"state.internal.workloads.kcp.dev/us-west1": "Sync",
+				}, nil, nil),
+			},
+			resourceToProcessLogicalClusterName: "",
+			resourceToProcessName:               "theDeployment",
+			workloadClusterName:                 "us-west1",
+			advancedSchedulingEnabled:           false,
+
+			expectError:         false,
+			expectActionsOnFrom: []clienttesting.Action{},
+			expectActionsOnTo: []clienttesting.Action{
+				getDeploymentAction("theDeployment", "test"),
+				updateDeploymentAction("test",
+					toUnstructured(t, changeDeployment(
+						deployment("theDeployment", "test", "", map[string]string{
+							"state.internal.workloads.kcp.dev/us-west1": "Sync",
+						}, nil, nil),
+						addDeploymentStatus(appsv1.DeploymentStatus{
+							Replicas: 15,
+						}))),
+					"status"),
+			},
+		},
+		"StatusSyncer upstream deletion": {
+			direction:              SyncUp,
+			upstreamLogicalCluster: "root:org:ws",
+			fromNamespace: namespace("kcp0124d7647eb6a00b1fcb6f2252201601634989dd79deb7375c373973", "",
+				map[string]string{
+					"state.internal.workloads.kcp.dev/us-west1": "Sync",
+				},
+				map[string]string{
+					"kcp.dev/namespace-locator": "{\"logical-cluster\":\"root:org:ws\",\"namespace\":\"test\"}",
+				}),
+			gvr: schema.GroupVersionResource{Group: "apps", Version: "v1", Resource: "deployments"},
+			fromResource: changeDeployment(
+				deployment("theDeployment", "kcp0124d7647eb6a00b1fcb6f2252201601634989dd79deb7375c373973", "", nil, nil, nil),
+				addDeploymentStatus(appsv1.DeploymentStatus{
+					Replicas: 15,
+				})),
+			toResources: []runtime.Object{
+				deployment("theDeployment", "test", "root:org:ws", map[string]string{
+					"state.internal.workloads.kcp.dev/us-west1": "Sync",
+				}, nil, nil),
+			},
+			resourceToProcessLogicalClusterName: "",
```

Comment:
what does it mean if this empty?

Reply:
In the SyncUp cases, the logicalClusterName is empty.